HW1/P4/P4_bfs.py

```diff
@@ -0,0 +1,23 @@
+def BFS(character, graph, accum):
+
+
+    # Commented out the max diameter assumptions
+    # diameter_max = 10
+    # diameter = 0
+    
+    tocheck = [character]
+    checked = []
+
+    while len(tocheck)!=0: # and diameter<diameter_max:
+        
+        # get the neighbors of those in the tocheck list and put them into a neighbors list
+        neighbors = graph.filter(lambda l: l[0] in tocheck).map(lambda l:l[1]).reduce(lambda a,b: a+[i for i in b if i not in a])
```

Comment:
The function provides the right results but I am worried about its execution time as you are using an action (reduce) for each stage to extract the neighbors. Maybe a version without any action which returns a rdd may be more efficient/